src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -1419,80 +1427,86 @@ void EmitBackreferenceConditional(RegexNode node)
                     // might try to backtrack to here, so output a backtracking map based on resumeAt.
-                    if (postIfDoneLabel != originalDoneLabel || postElseDoneLabel != originalDoneLabel)
+                    if (postYesDoneLabel != originalDoneLabel || postNoDoneLabel != originalDoneLabel)
                     {
                         // Skip the backtracking section
-                        // goto endRef;
-                        Br(backreferenceConditionalEnd);
+                        // goto endConditional;
+                        Br(endConditional);
 
                         Label backtrack = DefineLabel();
                         doneLabel = backtrack;
                         MarkLabel(backtrack);
 
                         // resumeAt = base.runstack[--stackpos];
-                        EmitRunstackPop();
+                        EmitStackPop();
                         Stloc(resumeAt);
 
-                        if (postIfDoneLabel != originalDoneLabel)
+                        if (postYesDoneLabel != originalDoneLabel)
                         {
                             // if (resumeAt == 0) goto postIfDoneLabel;
                             Ldloc(resumeAt);
                             Ldc(0);
-                            BeqFar(postIfDoneLabel);
+                            BeqFar(postYesDoneLabel);
                         }
 
-                        if (postElseDoneLabel != originalDoneLabel)
+                        if (postNoDoneLabel != originalDoneLabel)
                         {
                             // if (resumeAt == 1) goto postElseDoneLabel;
                             Ldloc(resumeAt);
                             Ldc(1);
-                            BeqFar(postElseDoneLabel);
+                            BeqFar(postNoDoneLabel);
                         }
 
                         // goto originalDoneLabel;
                         BrFar(originalDoneLabel);
                     }
                 }
 
-                if (postIfDoneLabel != originalDoneLabel || hasNo)
+                if (postYesDoneLabel != originalDoneLabel || noBranch is not null)
                 {
-                    MarkLabel(backreferenceConditionalEnd);
-                    if (!isAtomic && (postIfDoneLabel != originalDoneLabel || postElseDoneLabel != originalDoneLabel))
+                    MarkLabel(endConditional);
+                    if (!isAtomic && (postYesDoneLabel != originalDoneLabel || postNoDoneLabel != originalDoneLabel))
                     {
                         // if (stackpos + 1 >= base.runstack.Length) Array.Resize(ref base.runstack, base.runstack.Length * 2);
                         // base.runstack[stackpos++] = resumeAt;
-                        EmitRunstackResizeIfNeeded(1);
-                        EmitRunstackPush(() => Ldloc(resumeAt));
+                        EmitStackResizeIfNeeded(1);
```

Comment:
does any of the commented out equivalent C# need an update?

Reply:
eh, I haven't seen any yet.

Reply:
I don't think so?